pkg/api/errors/errors.go

```diff
@@ -136,30 +138,32 @@ func NewUnauthorized(reason string) error {
 }
 
 // NewForbidden returns an error indicating the requested action was forbidden
-func NewForbidden(kind, name string, err error) error {
+func NewForbidden(qualifiedResource unversioned.GroupResource, name string, err error) error {
```

Comment:
There are other reasons to be forbidden.  If you reference something
via an object ref the user is forbidden from using, a kind equal to
the object ref is needed in order to know *what* was forbidden.  We
use this heavily from OpenShift image streams.